### PR TITLE
[this pr is ready please stop ignoring it :)] Adds combat cybernetic implants to the uplink

### PR DESCRIPTION
**Pull request by silicons**
## DNM because balance concerns needs discussion etc etc

Adds combat implants to uplink, which would include an energy blade, flash, stun-arm, and medical beamgun, as well as a very costly but still discounted implant that has all 4 items in 1!
Energy blade costs 12 TC for nuclear operatives, 16 TC for traitors. Normal hand-held eswords are 8 TC, and have armor penetration, and block chance. The implant version do not have block chance, or armor penetration, but only make a quiet sparking on activation ( Unless I really should put something else as the sound)
Flash costs 6 TC for both.
Stun-arm costs 16 TC and is only for nuclear operatives.
Medical-Beamgun costs 20 TC (normal handheld item 16 TC) for nuclear operatives
An implant with all 4 items built into it is available to nuclear operatives for 48 TC.
:cl: Mekhi
rscadd: Traitors can now purchase the Energy Blade implant for 16 TC, and the Photon Projector implant for 6 TC.
rscadd: Nuclear Operatives can now purchase the Energy Blade implant for 12 TC, the Photon Projector implant for 6 TC, Medical Beam implant for 20 TC, Stun Arm implant for 16 TC, and an arm implant with all 4 for 48 TC which is a 10% off deal!
tweak: Photon Projector is just a fancy way of saying a flash that also works as a flashlight and never burns out unless it is exposed to electromagnetic burst radiation.
:cl:

Honestly the only balance concern I see with this is nukeops having what's basiclly stungloves and traitors getting a murderbone-tier weapon, even without the AP and block chance the esword implant is still really strong and has sharpness if that other PR goes in.

### CODE REVIEW

**Comment by optimumtact** [comment]
Fine with these as uplink items

File: code/modules/uplink/uplink_item.dm
Comment (Shadowlight213) [inline code/modules/uplink/uplink_item.dm]: OBJECT ORIENTED PROGRAMMING REEEE!
I said this on the previous PR. If the variable is the exact same as the parent, you don't set it on the child as well.

File: code/modules/uplink/uplink_item.dm
Comment (Shadowlight213) [inline code/modules/uplink/uplink_item.dm]: If this is only going to be in the operative uplink, there's no need to make the one use subtype for it, as the existing nuke op uplink implants use the standard autoimplanter. Just have this set to the implant itself. The existing code will load it into the normal autoimplanter.
Same with all the other op uplink only implants

File: code/modules/surgery/organs/autoimplanter.dm
Comment (Shadowlight213) [inline code/modules/surgery/organs/autoimplanter.dm]: See comment on the uplink datum.
You only really need to make the autoimplanter children for the esword and flash implants as those are the only ones that are in the traitor uplink.

**Review by Shadowlight213 — CHANGES_REQUESTED** [review]
Don't set identical variables on children as it defeats the point of object oriented programming.
Also, don't make one use subtypes for the nuke op only implants

File: code/modules/uplink/uplink_item.dm
Comment (kevinz000) [inline code/modules/uplink/uplink_item.dm]: Crap woops sowwy

**Comment by PKPenguin321** [comment]
G E T   S P R I T E S   F I R S T

Honestly the initial PR that added these shouldn't have been merged until it got sprites

**Comment by Shadowlight213** [comment]
Wait, these don't have sprites?

**Comment by PKPenguin321** [comment]
afaik no, they use default esword/stunbaton/etc sprites

**Comment by GunHog** [comment]
Come on, Mekhi, step it up!!

**Comment by PKPenguin321** [comment]
seriously though @maintainers stop merging this man's PRs until he gets real sprites

**Comment by optimumtact** [comment]
they're admin spawn weapons

**Comment by Shadowlight213** [comment]
Except this pr will make them not though?

**Comment by optimumtact** [comment]
that was in reference to the previous PR, not this one

**Comment by Niknakflak** [comment]
Adding weapons as admin weapons only is always the scapecoat to get bad things in game

**Comment by kevinz000** [comment]
Literally said last PR that this was being added

**Comment by ShadowDeath6** [comment]
what ever happened to the old ninja eblade sprite

File: code/modules/uplink/uplink_item.dm
Comment (kevinz000) [inline code/modules/uplink/uplink_item.dm]: Eh, what about adminspawn ones?

**Comment by xxalpha** [comment]
Alright. None of this is cybernetic, these should be augments.

**Comment by kevinz000** [comment]
@xxalpha We have cybernetic eguns and laser guns. Why not this? I don't see how an energyh blade projector could be """larger""" then a gun.

**Comment by xxalpha** [comment]
The answer is we shouldn't have those either. Do these things replace your limbs like the augments do?

**Comment by Lexorion** [comment]
4rlcb40
He can use those for the esword implant if he wants to,should probably change the names to hardlight blade or something similar though since eswords are supposed to be hot (=rip arm). The blade comes out of the wrist,it does not replace the hand so it should be fine as an implant. I might make something for the stunarm implant aswell.

**Comment by GunHog** [comment]
Does that have an animation? It needs one!

**Comment by Lexorion** [comment]
Yes,it's different from the esword(s) aswell.

**Comment by GunHog** [comment]
Awesome :D

**Comment by KorPhaeron** [comment]
Nuke ops with hidden stun batons sounds really boring

Not sure about others

**Comment by kevinz000** [comment]
Updating sprites on the other PR so it can be merged before this is considered if possible.

File: code/modules/uplink/uplink_item.dm
Comment (Shadowlight213) [inline code/modules/uplink/uplink_item.dm]: Admins can spawn their own implanters
Just do the change. Doing it this way is horribly inneficient

**Comment by PKPenguin321** [comment]
your changelog is shit and why do the inbuilt blades cost so much if you're nerfing them in https://github.com/tgstation/tgstation/pull/20762 ?

>no blocking or armor penetration
>basically die and lose sword to EMPs
>only upside is inventory convenience
>costs 1.5x as much TCs and 2x as much if you're a traitor

**Comment by ShadowDeath6** [comment]
yeah but if you get your dick kicked by sec there's no way for them to tell you have an esword arm

**Comment by KorPhaeron** [comment]
I think the sprites are super cool but this will likely never be very well balanced since it's pretty redundant vs an esword.

**Comment by optimumtact** [comment]
Replaces things that already exist and are better, but have no obvious downsides

:-1:

**Comment by ShadowDeath6** [comment]
>>>>better than them

but damage wise it's not better than them though

**Comment by Lexorion** [comment]
Well,we should still use my sprites for the adminsspawn version since it's already in the game ,just ping or memo me for the .dmi. We *could* make the DS or ERT (they already have eswords,right?) spawn with it,NTs finest deserve advanced tech and it gives the item some use without messing with the balance. Just a suggestion.

**Comment by KorPhaeron** [comment]
@Lexorion 

Bug me sometime tomomrow

**Pull request closed without merging**
